backend/app/controllers/spree/admin/payment_methods_controller.rb

```diff
@@ -49,16 +49,31 @@ def collection
       end
 
       def load_providers
-        @providers = Spree::PaymentMethod.providers.sort_by(&:name)
+        load_payment_method_types
       end
+      deprecate load_providers: :load_payment_method_types, deprecator: Spree::Deprecation
 
-      def validate_payment_method_provider
+      def load_payment_method_types
+        @payment_method_types = Rails.application.config.spree.payment_methods.sort_by(&:name)
+        # TODO: Remove `@providers` instance var once `load_providers` gets removed.
+        @providers = @payment_method_types
+      end
+
+      def validate_payment_provider
+        validate_payment_method_type
+      end
+      deprecate validate_payment_provider: :validate_payment_method_type,
+        deprecator: Spree::Deprecation
+
+      def validate_payment_method_type
         requested_type = params[:payment_method].delete(:type)
-        @payment_method_type = @providers.detect do |klass|
+        @payment_method_type = @payment_method_types.detect do |klass|
           klass.name == requested_type
         end
+        # TODO: Remove `@payment_method_type` instance var once `load_providers` gets removed.
+        @payment_method_type = @payment_method_type
```

Comment:
If you move the comment up to new line 70, we shouldn't really need this line, should we?